### PR TITLE
Add EventHandler Service.

### PR DESCRIPTION
**Pull request by eyurko**
Also update PythonCodeRunnerService and have some extra commits that still lingered despite me changing to a new branch.

 Annoying!

### CODE REVIEW

File: client/question/services/EventHandlerService.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that provides methods to create events on TIE's
+ * backend, if it exists.
+ */
+tie.factory('EventHandlerService', [
+  '$http', 'ServerHandlerService', function($http, ServerHandlerService) {
+    var FEEDBACK_CATEGORIES = [
```
Comment (seanlip) [inline client/question/services/EventHandlerService.js]: Declare externally as tie.constant, and then import it?

File: client/question/services/EventHandlerService.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      'SYNTAX', 'MATCHED BUGGY OUTPUT', 'UNMATCHED BUGGY OUTPUT', 'RUNTIME',
+      'PERFORMANCE', 'NO ERROR'];
+
+    return {
+      FEEDBACK_CATEGORIES: FEEDBACK_CATEGORIES,
+      /**
+       * Submits data to TIE's backend to create a SessionPauseEvent.
+       *
```
Comment (seanlip) [inline client/question/services/EventHandlerService.js]: Document args? Ditto elsewhere.

File: client/question/services/EventHandlerService.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+       */
+      createSessionPauseEvent: function(sessionId) {
+        if (ServerHandlerService.doesServerExist()) {
+          var data = {
+            sessionId: sessionId
+          };
+          $http.post('/ajax/admin/create_session_pause_event', data);
```
Comment (seanlip) [inline client/question/services/EventHandlerService.js]: Should be /ajax/event/create_session_pause_event, right? Ditto elsewhere.

Comment (eyurko) [reply client/question/services/EventHandlerService.js]: It's admin on the backend, which is probably incorrect, so I'll make a CL to fix that as well.

**Pull request merged**